packages/backend/src/peripherals/database/migrations/007_extend_provider_cache.ts

```diff
@@ -0,0 +1,36 @@
+/*
+                      ====== IMPORTANT NOTICE ======
+
+DO NOT EDIT OR RENAME THIS FILE
+
+This is a migration file. Once created the file should not be renamed or edited,
+because migrations are only run once on the production server. 
+
+If you find that something was incorrectly set up in the `up` function you
+should create a new migration file that fixes the issue.
+
+*/
+
+import { Knex } from 'knex'
+
+/**
+ * @notice Cache wipe
+ */
+export async function up(knex: Knex): Promise<void> {
+  // Wipe the table
+  await knex.table('provider_cache').truncate()
+
+  await knex.schema.alterTable('provider_cache', (table) => {
+    table.integer('chain_id').notNullable()
+    // Can be nullable in case of i.e getTransactions which is keyed by hash
+    // which in case of reorg will be discarded either way due to dangling tip
```

Comment:
Please explain what you mean here.

Reply:
The only thing that restrains us from making block_number nonNullable is the fact that transactions relies on hashes, not on the blocks. Just a double-check comment

Reply:
So they will stay in our database forever?

Reply:
I think even getTransaction should be cached with the block number that transaction is in. The block number is actually already fetched in the getTransaction response, so it needs to be fixed/passed in discovery.

But I still think it might be useful to allow null values for block number. I can imagine adding some meta-data values to the cache, like "hashOfLastCachedBlock" etc. which are not really cache values, but are useful information about the performance of cache, or some re-org protection data.

Reply:
Actually, after thinking about it for a while, I suggest we remove the ability for block number to be undefined. We have no clear use case for it, and it might cause future developers to assume that not passing block number is ok, while it should be almost never the case (e.g. even with getTransaction above).

This will require fixing it here and in discovery.

Reply:
Yup, I'm down for that (in the nearby future)